### ISSUE TITLE
Agenda Fatal Error - `TypeError - undefined is not an object (evaluating 'this[0][methodName]')`

### ISSUE THREAD

**Issue by tsalama** [wix/react-native-calendars#2006]
## Description

Upon opening an Agenda, there's an intermittent fatal error that happens one out of every four or five times coming from the `xdate` package. This started occuring after I upgraded RN to 0.68. Seems to be a race condition since it doesn't happen every time.

## Expected Behavior

Agenda opens without a fatal error causing an app crash.

## Observed Behavior

Fatal error is thrown causing app crash.

```
Error (Fatal): 'TypeError - undefined is not an object (evaluating 'this[0][methodName]')
```

## Environment

react-native-calendars: ^1.1288.1 (however, downgrading to 1.286.0 did not fix this issue.
react-native: 0.68
device: macOS Monterey 12.5 Xcode simulator

**Comment by nhinq2** [comment]
same issues for me
"react": "18.0.0",
 "react-native": "0.69.4",
 "react-native-calendars": "^1.1288.0",

**Comment by kizaonline** [comment]
We are experiencing the same error but with the CalendarList component:
"react": "18.0.0",
"react-native": "0.69.5",
"react-native-calendars": "^1.1288.2",

**Comment by stefanofgithub** [comment]
Same error with CalendarList:

"react": "18.0.0",
"react-native": "0.69.4",
"react-native-calendars": "^1.1287.0",

**Comment by vkourtis** [comment]
I had the same issue, but I think this might be fixed in [1.1289.0](https://github.com/wix/react-native-calendars/releases/tag/1.1289.0).

It used to happen consistently for a specific course of actions for me and not it doesn't.
Could be related to one of the CalendarList fixed that are mentioned in the changelog.

**Comment by genesisxyz** [comment]
Similar issue with the component ExpandableCalendar

`TypeError: undefined is not an object (evaluating 'd['get' + (useUTC ? 'UTC' : '') + methodSubjects[fieldIndex]]')`

I am using it with Expo SDK 46

**Comment by kizaonline** [comment]
This issue was resolved for us by updating to version `1.1289.0`
